macro/src/lib.rs

```diff
@@ -194,6 +209,24 @@ fn gen_params_invocation(param: &ast::Param, expanded_params: &mut Vec<Expr>) {
             let param = &param.name;
             expanded_params.push(parse2(quote!(#param.into())).unwrap());
         }
+        ast::TypeName::Function(in_types, out_type) => {
+            let cb_wrap_ident = Ident::new(&format!("{}_cb_wrap", param.name), Span::call_site());
+            let mut cb_param_list = vec![];
+            let mut cb_arg_type_list = vec![];
+            for (index, in_ty) in in_types.into_iter().enumerate() {
+                cb_param_list.push(Ident::new(&format!("arg{}", index), Span::call_site()));
+                cb_arg_type_list.push(in_ty.to_syn());
+            }
+            let cb_ret_type = out_type.to_syn();
+
+            let tokens = quote! {
+                move | #(#cb_param_list,)* | unsafe {
+                    std::mem::transmute::<unsafe extern "C" fn (*const c_void, ...) -> #cb_ret_type, unsafe extern "C" fn (*const c_void, #(#cb_arg_type_list,)*) -> #cb_ret_type>
```

Comment:
thought: `DiplomatCallback` could have a `get_callback()` function that returns an `impl Fn(..)`

It's nice to have the macro codegen be super simple